pkg/localprovider/localprovider_test.go

```diff
@@ -0,0 +1,158 @@
+package localprovider
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/managed-upgrade-operator/pkg/apis/upgrade/v1alpha1"
+	"github.com/openshift/managed-upgrade-operator/util/mocks"
+)
+
+const (
+	TEST_LOCAL_UPGRADECONFIG_NAME                 = "managed-upgrade-config"
+	TEST_LOCAL_UPGRADECONFIG_VERSION              = "4.7.11"
+	TEST_LOCAL_UPGRADECONFIG_CHANNELGROUP         = "stable"
+	TEST_LOCAL_UPGRADECONFIG_TIME                 = "2021-06-03T00:00:00Z"
+	TEST_LOCAL_UPGRADECONFIG_PDB_TIME             = 60
+	TEST_LOCAL_UPGRADECONFIG_UPGRADETYPE          = "OSD"
+	TEST_LOCAL_UPGRADECONFIG_CAPACITY_RESERVATION = true
+	TEST_OPERATOR_NAMESPACE                       = "test-managed-upgrade-operator"
+)
+
+var _ = Describe("Local Provider", func() {
+	var (
+		mockCtrl          *gomock.Controller
+		mockKubeClient    *mocks.MockClient
+		provider          *localProvider
+		upgradeConfigList v1alpha1.UpgradeConfigList
+		upgradeConfigSpec v1alpha1.UpgradeConfigSpec
+	)
+
+	BeforeEach(func() {
+		mockCtrl = gomock.NewController(GinkgoT())
+		mockKubeClient = mocks.NewMockClient(mockCtrl)
+		provider = &localProvider{
+			client:  mockKubeClient,
+			cfgname: TEST_LOCAL_UPGRADECONFIG_NAME,
+		}
+
+		upgradeConfigList = v1alpha1.UpgradeConfigList{
+			Items: []v1alpha1.UpgradeConfig{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "uc1",
+					},
+					Spec: v1alpha1.UpgradeConfigSpec{
```

Comment:
Can we just use the defined spec contains, like:
```
Spec: upgradeConfigSpec,
```
If they have the same values.

Reply:
Nice suggestion. Would be cleaner. Done.